Ruko destroys meth

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

* Slows down the rate at which meth is metabolized, allowing for smaller doses to provide a longer effect period
* Makes the addiction side effects of meth non-lethal (No more toxin damage at stage 4)
* Makes the addiction side effects of meth less annoying (lower chance of forced emotes)
* Makes the addiction and overdose thresholds for meth dynamic
* Makes positive meth effects build over time instead of behaving like an instant combat-ready adrenal stim.
* Removes sleep immunity from meth
* Reduced the explosive power of meth to equal potassium/water - players wanting explosives should make dedicated explosive reagents for the biggest boom. 
* Makes all drugs (/datum/reagent/drug) purge Mannitol from the user's system (alternative to the idea presented in #7198).
* ~~Removes the run tests for meth increasing speed on_mob_metabolize() and on_mob_end_metabolize() since this behavior is being changed and meth is not guaranteed to affect speed.~~
* Implements Badminordrazine to take the place of meth in the above reagent tests. It also doubles as a new admin smite and floorpill chem. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Meth is the single most powerful reagent in the game, beaten only by nitroglycerin in terms of explosive power, and beaten by pretty much nothing in terms of being a combat stim. This makes the drug a more nuanced "jack of trades" choice when compared to other options. It remains powerful as a combat stim to those willing to brave the (now reduced) side effects of addiction and overdosing, especially if taken regularly or ahead of getting into an engagement. 

As for the mannitol part, I mostly agree with the points made in #7198, just not the execution. Being able to easily disregard the downside to something means the downside may as well not resist. My solution is less nuclear and doesn't totally prevent the use of mannitol while on drugs, but greatly diminishes its usefulness. 

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

100u meth explosion after the nerf, size (1, 2, 4, 0)
![image](https://user-images.githubusercontent.com/9547572/177674820-b521cef2-5ec5-4258-b4d5-688017216631.png)

Showing the dynamic overdose threshold taking effect after continuous microdosing:
![image](https://user-images.githubusercontent.com/9547572/177677448-d7b84a57-a135-4bb2-bcd9-0c5f71e60c3c.png)

</details>

## Changelog
:cl:
balance: All of meth's positive effects are now built over time, with movement speed being capped at 50 seconds and the stun resistance and stamina restoration effects having no cap, but starting low. Sleep immunity has been removed. 
balance: Meth explosions are now equal to potassium + water so that dedicated explosive reagents are worth making
tweak: Meth metabolizes 33% slower, so a smaller dose will last longer. 
tweak: Meth now has dynamic addiction and overdose thresholds instead of static ones. 
tweak: Meth's addiction side effects are now completely non-lethal and the forced emotes proc far less often as well.
tweak: All drugs purge mannitol from the user's system. Recovering from brain damaging drugs requires a sober period
add: Badminordrazine is a new reagent which can be used by staff for smites and also has a small chance of appearing in floor pills. It inflicts any of ten random effects every six seconds it is being processed. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Ruko has ruined meth!

Rukofamicom committed:
Update drug_reagents.dm

Rukofamicom committed:
Update drug_reagents.dm

Rukofamicom committed:
Update drug_reagents.dm

Rukofamicom committed:
Revert "Update drug_reagents.dm"

This reverts commit 2b4ed5989065d8b5798e24466dd6db2af04d56c8.

Rukofamicom committed:
Revert "Update drug_reagents.dm"

This reverts commit 44e5f3fc4fca89f358699fb99a180635d2779f83.

francinum commented on code/modules/reagents/chemistry/reagents/drug_reagents.dm:
```diff
@@ -191,40 +195,48 @@
+	M.adjustStaminaLoss(-current_cycle, FALSE)
+	M.drowsyness = max(0,M.drowsyness-current_cycle)
+	M.Jitter(2)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, min(1, current_cycle*0.05))
```
This will make the maximum brain damage 1 per cycle, is this intentional?

Rukofamicom replied:
Yes, I figure if I'm making the positive effects onset slowly, the brain damage should as well. Brain damage peaks at 1 per tick just before movespeed reaches its maximum like this. 

Brain damage has the potential to be fairly debilitating, and I'm removing the ability to effectively stack drugs with mannitol so it will seem even moreso to the people used to doing this.

Rukofamicom committed:
Removes meth check

PowerfulBacon commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -167,7 +167,7 @@
 	required_temp = 380 //slightly above the meth mix time.
 	required_reagents = list(/datum/reagent/drug/methamphetamine = 1)
-	strengthdiv = 6
+	strengthdiv = 10
```
I think this should be stronger than a 2 base chemical explosion, considering its hard to make

Rukofamicom replied:
I think it's already strong enough just because it's such a multi-purpose chem. It's a single chem that can both be used to immunize yourself to sec tools and also offensively attack others if used in grenades. Pot-water explosions aren't instant-lethal (usually), but they're still explosions which are powerful stuns if nothing else. 

It's also especially useful that meth isn't immediately identified as being an explosive, even if your intention is to use it for that purpose... unless the finished explosive has already been assembled.

Rukofamicom replied:
~~Regardless of the points above, will adjust this as requested if you would like to insist it be turned up a bit more.~~

See most recent comment by me. Due to time constraints PR should be accepted as-is or closed if this isn't acceptable.

PowerfulBacon commented on code/modules/unit_tests/metabolizing.dm:
```diff
@@ -11,11 +11,9 @@
 	user.Life()
 	TEST_ASSERT(!user.reagents.has_reagent(meth), "User still has meth in their system when it should've finished metabolizing")
-	TEST_ASSERT(!user.has_movespeed_modifier(/datum/reagent/drug/methamphetamine), "User still has movespeed modifier despite not containing any more meth")
 
```
Update this test to use a different reagent that does something to verify the reagent is working

Rukofamicom replied:
👍

Let's see what I can find for it.

Rukofamicom replied:
I have implemented a new "fun" type chem called Badminordrazine to use for the check. It also doubles as an adminbus chem to smite players with that isn't so over-the-top it can't see use. 

*It can also show up in floorpills just like adminordrazine*

Dejaku51 replied:
just change it to crank and check if they have TRAIT_NOBLOCK. no need to create new reagent just for that

Rukofamicom committed:
Implements Badminordrazine

Also reintroduces the test I removed in the last commit, and replaces meth with Badminordrazine.

PowerfulBacon commented on code/modules/reagents/chemistry/reagents/drug_reagents.dm:
```diff
@@ -191,40 +250,48 @@
-	L.add_movespeed_modifier(type, update=TRUE, priority=100, multiplicative_slowdown=-0.5, blacklisted_movetypes=(FLYING|FLOATING))
-	ADD_TRAIT(L, TRAIT_SLEEPIMMUNE, type)
 	ADD_TRAIT(L, TRAIT_NOBLOCK, type)
+	addiction_threshold = rand(0, 10) //Highly addictive substances are risky. You never know exactly when you'll be addicted
+	overdose_threshold = rand(10, 20)
```
Way too low, especially since it decreases every tick

Rukofamicom replied:
You will never hit overdose or addiction in the middle of metabolization, only immediately after taking more. 

It reduces every tick but by less than the metabolization rate.
